src/js/editor/editor.js

```diff
@@ -630,11 +629,24 @@ class Editor {
         this.handleNewline(event);
         break;
       case key.isPrintable():
-        let range = this.cursor.offsets;
-        if (this.cursor.hasSelection()) {
-          let nextPosition = this.run(postEditor => postEditor.deleteRange(range));
+        let { offsets: range } = this.cursor;
+        let { isCollapsed } = range;
+        let nextPosition = range.head;
+        this.run(postEditor => {
+          if (!isCollapsed) {
+            nextPosition = postEditor.deleteRange(range);
+          }
+          if (!range.head.section.isCardSection && key.isTab()) {
+            nextPosition = postEditor.insertText(nextPosition, '\t');
```

Comment:
the conditional would read more cleanly (imo) is the `isCardSection` is last (like the `ifCollapsed` conditional a few lines below this):
`if (key.isTab() && !range.head.section.isCardSection)`

Also the `'\t'` should probably be a constant: `TAB_CHARACTER` or similar.